docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,632 @@
+`TEXT` will not be truncated. It's generally more efficient to store and perform operations on a `CHAR` or `VARCHAR` 
+so if possible, try to use those. Columns like usernames, first/last names, emails, website URLs, and others, whose 
+length generally won't be longer than a certain length should be stored as that.
+
+#### INTEGER/BIGINTEGER
+
+The `INTEGER` or `INT` type is an integer with a range of [-2147483648, 2147483647]. The `BIGINTEGER`/`BIGINT`
+is an integer value with an extended range of [-9223372036854775808, 9223372036854775807] (wow that was terrible to type
+out). 
+
+#### DECIMAL
+
+The `DECIMAL` type is similar to a double/float in other programming languages. You can usually specify the precision
+and scale of the column by writing it like this: `DECIMAL(19, 2)`. 
+
+**Precision** defines the total count of significant digits in the number (total number of digits on *both sides* of the
+decimal point).
+**Scale** defines the count of digits to the right of the decimal.
+
+The range of this data type is *up to 131072 digits before the decimal point; up to 16383 digits after the decimal 
+point*.
+
+#### TIMESTAMP
+
+The `TIMESTAMP` type is one of the date/time types commonly seen in SQL databases. There's also a `TIMESTAMP` type with
+a time zone, if that suits your needs better. 
+
+#### BOOLEAN/TINYINTEGER
+
+The `BOOLEAN` is a type which is available in PostgreSQL. In many other DBs, `TINYINTEGER` or `TINYINT` is commonly used
+instead (they are the same thing). 
+
+#### ENUM
+
+The `ENUM` type allows you to specify a small set of possible values for a column. In PostgreSQL, enums are created as
+a new datatype and then used in tables afterward. In other DB management systems, enums are usually just written as 
+you're declaring a column type.
+
+Here's an example of creating an enum type, which is shown used in a table later on:
+```sql
+CREATE TYPE personality AS ENUM ('nice', 'mean', 'neither');
+```
+
+### Column Modifiers
+
+Column modifiers are special modifiers you add when declaring a column in a table which give it extra properties. Here
+are a few of the common ones (some that you may need soon too!).
+
+#### PRIMARY KEY
+
+The `PRIMARY KEY` modifier declares a column or combination of columns as the ID for a given row. The database also uses
+a `PRIMARY KEY` to speed up some search functions that use those fields (see 
+[Indexes](https://www.postgresql.org/docs/9.1/indexes.html) if you're interested ). If used, all primary 
+keys must be unique on the given table. This column is also not nullable.
+
+#### NOT NULL
+
+The `NOT NULL` modifier lets the database know that a column should never accept null values. You should try to use this
+where you can, since it can help with preventing nulls from being entered where you're not expecting them.
+
+#### AUTO INCREMENT
+
+The `AUTO INCREMENT` modifier lets the database that it should automatically enter this column as the default value
```

Comment:
maybe mention that this will be helpful for PRIMARY KEY because you won't need to populate the table

Reply:
Added a comment to that and the `SERIAL` type